bitcoin/tx.c

```diff
@@ -514,11 +529,16 @@ struct bitcoin_tx *bitcoin_tx_with_psbt(const tal_t *ctx, struct wally_psbt *psb
 					   psbt->tx->num_outputs,
 					   psbt->tx->locktime);
 	wally_tx_free(tx->wtx);
-	tx->wtx = psbt_finalize(tx, psbt, false);
+
+	psbt_finalize(psbt);
```

Comment:
iirc we don't want to actually finalize the `psbt` here as that'll update the passed in psbt. we want to extract the final_tx, which requires finalization but ideally on a different structure than the passed in one.